test/test_rosdep_gbpdistro_support.py

```diff
@@ -52,7 +52,9 @@ def test_url_constants():
 def test_get_gbprepo_as_rosdep_data():
     from rosdep2.rosdistrohelper import get_index
     from rosdep2.gbpdistro_support import get_gbprepo_as_rosdep_data
-    distro = sorted(get_index().distributions.keys())[0]
+    distro = sorted(
+            d for d, info in get_index().distributions.items()
+            if info['distribution_type'] == 'ros1')[0]
```

Comment:
This change would generate an empty list causing the `[0]` to index error immediately after. Is your goal to allow this to work without a specified distribution_type or to fail explicitly rather than with arbitrary errors?

Reply:
`info['distribution_type']` might fail with a key error where my proposal would not.

If there is not a single ROS 1 distribution the result is the same as before where it tries to access the first item on an empty list.

Reply:
> info['distribution_type'] might fail with a key error where my proposal would not.

I don't see any significant difference between a KeyError and an IndexError in this case but it's an easy request to satisfy.